services_app/src/androidTest/java/org/opendatakit/database/service/OdkDatabaseServiceImplTest.java

```diff
@@ -1563,17 +1564,21 @@ private static List<KeyValueStoreEntry> makeMetadata() {
       return l;
    }
 
-   private static File makeTemp(String id) throws Exception {
+   private static DocumentFile makeTemp(String id) throws Exception {
       File instanceFolder = new File(
-          ODKFileUtils.getInstanceFolder(APPNAME, TEA_HOUSES_TBL_NAME, id));
+              ODKFileUtils.getInstanceFolder(APPNAME, TEA_HOUSES_TBL_NAME, id));
       if (!instanceFolder.exists() && !instanceFolder.mkdirs()) {
          throw new Exception("Should have been able to create " + instanceFolder.getPath());
       }
       assertTrue(instanceFolder.exists());
-      FileOutputStream f = new FileOutputStream(new File(instanceFolder.getPath() + "/temp"));
-      f.write(new byte[] { 97, 98, 99 });
+      File tempFile = new File(instanceFolder, "temp");
```

Comment:
@wbrunette what do you think, please. Although he converted the File object to a DocumentFile in line 1580

Reply:
You are correct, 

There are other ways to create a DocumentFile such as [fromSingleUri](https://developer.android.com/reference/androidx/documentfile/provider/DocumentFile#fromSingleUri(android.content.Context,android.net.Uri))

OutputStream outputStream = context.getContentResolver().openOutputStream(newFile.getUri());
                if (outputStream != null) {
                    outputStream.write(content.getBytes());
                    outputStream.close();
}